src/daft-connect/src/translation/logical_plan.rs

```diff
@@ -29,42 +29,74 @@ mod to_df;
         plan_id: usize,
         schema: Arc<Schema>,
         tables: Vec<Table>,
     ) -> eyre::Result<LogicalPlanBuilder> {
-        let partition_key = uuid::Uuid::new_v4().to_string();
+        let runner = self.session.get_runner()?;
 
-        let pset = Arc::new(MicroPartitionSet::from_tables(plan_id, tables)?);
+        match runner {
+            Runner::Ray => {
+                let mp =
+                    MicroPartition::new_loaded(tables[0].schema.clone(), Arc::new(tables), None);
+                Python::with_gil(|py| {
+                    // Convert MicroPartition to a logical plan using Python interop.
+                    let py_micropartition = py
+                        .import(intern!(py, "daft.table"))?
+                        .getattr(intern!(py, "MicroPartition"))?
+                        .getattr(intern!(py, "_from_pymicropartition"))?
+                        .call1((PyMicroPartition::from(mp),))?;
 
-        let PartitionMetadata {
-            num_rows,
-            size_bytes,
-        } = pset.metadata();
-        let num_partitions = pset.num_partitions();
+                    // ERROR:   2: AttributeError: 'daft.daft.PySchema' object has no attribute '_schema'
+                    let py_plan_builder = py
+                        .import(intern!(py, "daft.dataframe.dataframe"))?
+                        .getattr(intern!(py, "to_logical_plan_builder"))?
+                        .call1((py_micropartition,))?;
+                    let py_plan_builder = py_plan_builder.getattr(intern!(py, "_builder"))?;
+                    let plan: PyLogicalPlanBuilder = py_plan_builder.extract()?;
 
-        self.psets.put_partition_set(&partition_key, &pset);
+                    Ok::<_, eyre::Error>(dbg!(plan.builder))
+                })
+            }
+            Runner::Native => {
+                let partition_key = uuid::Uuid::new_v4().to_string();
 
-        let cache_entry = PartitionCacheEntry::new_rust(partition_key.clone(), pset);
+                let pset = Arc::new(MicroPartitionSet::from_tables(plan_id, tables)?);
 
-        Ok(LogicalPlanBuilder::in_memory_scan(
-            &partition_key,
-            cache_entry,
-            schema,
-            num_partitions,
-            size_bytes,
-            num_rows,
-        )?)
+                let PartitionMetadata {
+                    num_rows,
+                    size_bytes,
+                } = pset.metadata();
+                let num_partitions = pset.num_partitions();
+
+                self.session.psets.put_partition_set(&partition_key, &pset);
+
+                let cache_entry = PartitionCacheEntry::new_rust(partition_key.clone(), pset);
+
+                Ok(LogicalPlanBuilder::in_memory_scan(
+                    &partition_key,
+                    cache_entry,
+                    schema,
+                    num_partitions,
+                    size_bytes,
+                    num_rows,
+                )?)
+            }
+        }
```

Comment:
Is the reason why the implementations are different because we would like to avoid calling into Python for the native runner?

Reply:
yeah for the most part.